app/Jobs/DeleteWikiDbJob.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace App\Jobs;
+
+use App\WikiDb;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use PDOException;
+use App\Wiki;
+use Carbon\Carbon;
+
+/**
+ * Prepends the MW Database, User with `deleted_` prefix and deletes WikiDB relation for a wiki 
+ */
+class DeleteWikiDbJob extends Job implements ShouldBeUnique
+{
+    private $wikiId;
+
+    /**
+     * @return void
+     */
+    public function __construct( int $wikiId )
+    {
+        $this->wikiId = $wikiId;
+    }
+
+    /**
+     * The unique ID of the job.
+     *
+     * @return string
+     */
+    public function uniqueId()
+    {
+        return strval($this->wikiId);
+    }
+
+    /**
+     * @return void
+     */
+    public function handle()
+    {
+
+        $wiki = Wiki::withTrashed()->where(['id' => $this->wikiId])->first();
+
+        if( !$wiki ) {
+            $this->fail(new \RuntimeException("Wiki not found for {$this->wikiId}"));
+            return;
+        }
+
+        if( !$wiki->deleted_at ) {
+            $this->fail(new \RuntimeException("Wiki {$this->wikiId} is not marked for deletion."));
+            return;
+        }
+
+        $wikiDB = WikiDb::whereWikiId( $this->wikiId )->first();
+
+        if( !$wikiDB ) {
+            $this->fail(new \RuntimeException("WikiDb not found for {$this->wikiId}"));
+            return;
+        }
+
+        $conn = DB::connection('mw');
+        if (! $conn instanceof \Illuminate\Database\Connection) {
+            $this->fail(new \RuntimeException('Must be run on a PDO based DB connection'));
+            return;
+        }
+
+        $pdo = $conn->getPdo();
+        $sm = $conn->getDoctrineSchemaManager();
+        $tableNames = $sm->listTableNames();
+
+        $timestamp = Carbon::now()->timestamp;
+        $deletedDatabaseName = "deleted_{$timestamp}_{$this->wikiId}";
+
+        try {
+            $pdo->beginTransaction();
+
+            // Create the new database
+            $pdo->exec(sprintf('CREATE DATABASE %s', $deletedDatabaseName));
+
+            // iterate over each table and replace the prefix
+            // and move it to the deleted database by using RENAME TABLE
+            foreach($tableNames as $table) {
```

Comment:
At this point $tableNames is all tables in all databases on the `mw` connection as far as I can tell?
So in this loop we would actually expect `$replacedCount !== 1` to happen, as this will be for all of the tables that do not belong to the current wiki we are trying to delete?

Rather than doing this with only strings perhaps we could use https://www.doctrine-project.org/projects/doctrine-dbal/en/latest/reference/schema-manager.html#listtabledetails ?

This is probably a case that can be added to the test case

At some point looping through ALL tables on the DB might end up creating a bit of overhead we want to avoid.